### ISSUE TITLE
logging miscategorizes logs from GKE

### ISSUE THREAD

**Issue by dennismartensson** [GoogleCloudPlatform/google-cloud-node#2125]
We did try the new logging-bunyan with our bunyan logging node.js application today. 

When we put it under some load the cpu when to 100% and left the application crashing. It is logging fine with just bunyan. 

If you are still having issues, please be sure to include as much information as possible:

#### Environment details

  - OS: Docker kubernetes from node:6.9
  - Node.js version: node 6.9
  - npm version:
  - google-cloud-node version: 0.1.0

#### Steps to reproduce

  1. require `google-cloud`
  2. Was setting it up like this:

```js
const Logger = require('bunyan')
const loggingBunyan = require('@google-cloud/logging-bunyan')({
  keyFilename: 'keyfile.json',
  projectId: 'project-id',
})


const log = new Logger({
  name: 'name',
  src: true,
  streams: [
    loggingBunyan.stream('info'),
  ] })
```

Thanks!

**Comment by dennismartensson** [comment]
Hi @ofrobots and @stephenplusplus let me know how I can help.

Do you think that 0.2.0 fixes some of this?

Thank you

Dennis

**Comment by ofrobots** [comment]
Hi @dennismartensson. 0.2.0 fixes issues with objects with circular references. This would have manifested as high CPU usage and a stack overflow. Can you try 0.2.0 and report back?

**Comment by dennismartensson** [comment]
@ofrobots I will do that.

**Comment by dennismartensson** [comment]
So @ofrobots now we are trying it a bit under some load. The cpu is now better but we get no logs at all on https://console.cloud.google.com/logs

```
const Logger = require('bunyan')

const loggingBunyan = require('@google-cloud/logging-bunyan')({
  keyFilename: 'keyfile.json',
  projectId: 'project-id',
})

const log = new Logger({
  name: 'name',
  src: true,
  streams: [
    loggingBunyan.stream('info'),
  ] })
```

But there is no loggs.

**Comment by dennismartensson** [comment]
Did also try to change it to be like the example but its not showing any loggs in the logs viewer...

What can I do to help resolve this?

```
const bunyan = require('bunyan')

const loggingBunyan = require('@google-cloud/logging-bunyan')({
  keyFilename: './keyfile.json',
  projectId: 'project-id',
})

const log = bunyan.createLogger({
  name: 'name',
  streams: [
    loggingBunyan.stream('info'),
  ],
})
```

**Comment by ofrobots** [comment]
@dennismartensson: I'm looking into this now – I'm trying to run your example almost verbatim, but against my project, and I do see the log showing up. It is curious why it is not working in your case. What version of Node.js are you using?

**Comment by ofrobots** [comment]
This is the test I am running:

```
const bunyan = require('bunyan')

const loggingBunyan = require('@google-cloud/logging-bunyan')({
  keyFilename: './keyfile.json',
  projectId: 'ofrobots-test-15',
})

const log = bunyan.createLogger({
  name: 'name',
  streams: [
    loggingBunyan.stream('info'),
  ],
});

log.info('hello world');
```

**Comment by ofrobots** [comment]
Another question: which environment are you running on? GAE, GKE, GCE, Local?

**Comment by ofrobots** [comment]
Just in case this might be the issue: when looking for logs in the UI, make sure you have the correct log filters/categories setup. Example:

![image](https://cloud.githubusercontent.com/assets/79017/24227673/147fd176-0f2c-11e7-885a-19909e5fa62f.png)

**Comment by dennismartensson** [comment]
We are running on GKE:

OS: Docker kubernetes from node:6.9
Node.js version: node 6.9

Yes think I have the correct filter are showing all loggs for the cluster.

Thank you

**Comment by dennismartensson** [comment]
There seams to be a memory leak:

<img width="553" alt="screen shot 2017-03-23 at 12 36 17 pm" src="https://cloud.githubusercontent.com/assets/662057/24246906/b5a42f14-0fc9-11e7-937a-4006a62b4876.png">
<img width="1153" alt="screen shot 2017-03-23 at 12 34 39 pm" src="https://cloud.githubusercontent.com/assets/662057/24246912/bd0ae4dc-0fc9-11e7-8aeb-008e070414f2.png">

It goes down and stay leveled when the stream is removed.

Thank you

Dennis

**Comment by ofrobots** [comment]
I'm investigating this on a GKE cluster. Do you see anything in the stdout logs for your pods? `kubectl logs <name-of-one-of-your-pods>`.

**Comment by ofrobots** [comment]
I can reproduce this issue. It seems that we are reporting the logs from k8s pods as a 'gce_instance' resource. I suspect that the logs are showing up miscategorized.

Can you check with the following filters to see if your logs are there:

![image](https://cloud.githubusercontent.com/assets/79017/24282649/18093b1e-101e-11e7-9a3e-1a19c1d7835a.png)

We will get this fixed and then follow up on the potential memory leak.

**Comment by dennismartensson** [comment]
You are correct there is loggs with the filter you showed so its a strange category.

For the memory problem we where seeing the problem frequently when having the logger and stream configured when removed we do not see the problem.

How can we help solve this? The load we put is around  100rps per pod and 5 pods. We use restify's requestLogger in addition to our normal logging. 

Any thing I can do to help?

Thank you

Dennis

**Comment by ofrobots** [comment]
@dennismartensson I have extracted the memory leak bug into a separate bug so that we can track independently: https://github.com/GoogleCloudPlatform/google-cloud-node/issues/2135.

**Comment by dennismartensson** [comment]
@ofrobots cool I will keep a eye on that, Any ETA on a fix for that and the category fix? We are happy to do a update and put some traffic on to it again for testing purposes.

Thank you for the quick and responsiveness in getting stuff out there!

**Comment by ofrobots** [comment]
For the memory leak, we want to understand the leak first and we can go from there. We should be able to put together the category fix shortly.

**Comment by ofrobots** [comment]
@dennismartensson With https://github.com/GoogleCloudPlatform/google-cloud-node/pull/2163 released via 0.3.0 of `@google-cloud/logging-bunyan` you should see the logs going against the 'GKE Container' resource rather than GCE.

We are still investigating the memory leak in #2135, so I wouldn't recommend jumping on 0.3.0 just yet, except for experimentation.

I am going to close this issue, but feel free to re-open if the categorization doesn't match your expectations.

**Comment by dennismartensson** [comment]
Hi @ofrobots Thanks for digging in to this!

Looking forward to when the memory leak is fixed and we can start to use it again
